src/pages/main/components/selected-dataset/build-flow/components/wizard/wizard.store.ts

```diff
@@ -77,50 +77,47 @@ class WizardStore {
         }
       },
     )
-
-    reaction(
-      () => datasetStore.dsInfo.isLoading,
-      isLoading => {
-        !isLoading && this.defineAndSetNewScenario()
-      },
-    )
   }
 
   public toggleIsWizardVisible(value: boolean) {
     this.isWizardVisible = value
   }
 
   public get secondaryDatasets(): string[] | undefined {
-    return dirinfoStore.dirinfo.data?.dsDict[datasetStore.datasetName]
-      ?.secondary
+    return dirinfoStore.dirInfoData?.dsDict[datasetStore.datasetName]?.secondary
   }
 
   public getSecondaryDsByParentDs(parentDs: string): string[] | undefined {
     return dirinfoStore.dirinfo.data?.dsDict[parentDs]?.secondary
   }
 
-  public setScenario(scenario: IWizardScenario[]) {
+  public setScenario(
+    scenario: IWizardScenario[],
+    shouldSaveInHistory: boolean,
```

Comment:
make shouldSave have default value true, and remove second parameter in calling function where it is unnecessary